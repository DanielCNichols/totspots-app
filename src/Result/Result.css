.state {
  text-transform: uppercase;
}

.city {
  text-transform: capitalize;
}

.type {
  text-transform: capitalize;
}

.result {
  word-wrap: break-word;
  width: 90%;
  line-height: 1.5rem;
  border: 2px solid #064789;
  border-radius: 3px;
  box-shadow: 0 0 5px rgba(0, 0, 0, 0.7);
  margin: 10px auto;
}

.resultPhoto {
  margin: 0;
  height: 200px;
}

.resultPhoto img {
  width: 100%;
  height: 200px;
}

.result h3 {
  margin: 0;
  color: white;
}

.result_rating {
  background-color: #f14038;
  width: 100%;
  padding: 5px 15px;
  margin: 0;
  display: flex;
  justify-content: space-between;
}

.result_address {
  padding: 10px;
  width: 90%;
  margin: 0 auto;
}

.price_span {
  color: green;
  padding: 5px;
}

.rating {
  color: gold;
}

.result p {
  margin: 0;
}

.price_span,
.NA {
  min-width: 70%;
  display: inline-block;
  font-size: 1.2em;
  text-align: right;
  margin-left: auto;
}

.result button {
  display: none;
}

.NA > span {
  color: black;
}
@media screen and (min-width: 450px) {
  .result {
    min-width: 245px;
    width: 90%;
    font-size: 1.2em;
  }

  .result_rating {
    font-size: 1.1em;
    padding: 10px;
  }

  .result p {
    margin: 10px 0;
  }

  .result_address {
    width: 90%;
    margin: 10px auto;
  }

  .result button {
    display: block;
    margin: 10px auto;
    height: 35px;
    width: 40%;
    border-radius: 3px;
    background-color: transparent;
    border-color: #064789;
  }

  .result button:hover {
    cursor: pointer;
    color: white;
    background-color: #064789;
  }
}
