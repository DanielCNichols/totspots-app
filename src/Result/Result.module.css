* {
  box-sizing: border-box;
}

.result {
  width: 100%;
  display: grid;
  grid-template-columns: [photo] 150px 1fr;
  grid-template-rows: repeat(4, auto);
  margin: 0px auto;
  border-top: 2px solid #e7e7e7;
  /* border-bottom: 2px solid #e7e7e7; */
  padding: 25px 5px;
}

.result h4,
.ratings,
.address,
.types {
  padding: 0 15px 0 10px;
  grid-column: 2;
}

.result img {
  grid-column: photo;
  grid-row: 1 / 5;
  width: 100%;
  border-radius: 5px;
  max-height: 100px;
}

.result h4 {
  grid-column: 2;
  grid-row: 1;
  text-align: left;
  margin: 0;
  font-size: 1.1em;
}

.ratings {
  grid-column: 2;
  grid-row: 2;
  display: flex;
  justify-content: space-between;
}

.address {
  grid-column: 2;
  grid-row: 3;
  word-wrap: break-word;
}

.types {
  grid-column: 2;
  grid-row: 4;
  text-align: left;
}

.types span {
  display: inline-block;
  margin-right: 5px;
}

/* .result { */
/*   display: block; */
/*   word-wrap: break-word; */
/*   width: 100%; */
/*   line-height: 1.5rem; */
/*   border-bottom: 2px solid #e7e7e7; */
/*   padding: 25px 0; */
/*   margin: 10px 0; */
/* } */

/* .result img { */
/*   margin: 0 5px; */
/*   max-width: 150px; */
/*   border-radius: 5px; */
/*   float: left; */
/*   margin-right: 5px; */
/* } */

/* .result h3 { */
/*   margin: 0; */
/*   color: white; */
/* } */

/* .result_rating { */
/*   background-color: #f14038; */
/*   width: 100%; */
/*   padding: 5px 15px; */
/*   margin: 0; */
/*   justify-content: space-between; */
/* } */

/* .result_address { */
/*   padding: 10px; */
/*   width: 90%; */
/*   margin: 0 auto; */
/* } */

/* .price_span { */
/*   color: green; */
/*   padding: 5px; */
/* } */

/* .rating { */
/*   color: gold; */
/* } */

/* .result p { */
/*   margin: 0; */
/* } */

/* .price_span, */
/* .NA { */
/*   min-width: 70%; */
/*   display: inline-block; */
/*   font-size: 1.2em; */
/*   text-align: right; */
/*   margin-left: auto; */
/* } */

/* .result button { */
/*   display: none; */
/* } */

/* .NA > span { */
/*   color: black; */
/* } */
/* @media screen and (min-width: 450px) { */
/*   .result { */
/*     min-width: 245px; */
/*     width: 90%; */
/*     font-size: 1.2em; */
/*   } */

/*   .result_rating { */
/*     font-size: 1.1em; */
/*     padding: 10px; */
/*   } */

/*   .result p { */
/*     margin: 10px 0; */
/*   } */

/*   .result_address { */
/*     width: 90%; */
/*     margin: 10px auto; */
/*   } */

/*   .result button { */
/*     display: block; */
/*     margin: 10px auto; */
/*     height: 35px; */
/*     width: 40%; */
/*     border-radius: 3px; */
/*     background-color: transparent; */
/*     border-color: #064789; */
/*   } */

/*   .result button:hover { */
/*     cursor: pointer; */
/*     color: white; */
/*     background-color: #064789; */
/*   } */
/* } */
