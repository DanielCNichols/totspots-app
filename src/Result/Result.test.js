
import React from 'react';
import ReactDOM from 'react-dom'
import Result from './Result.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'


const venue =  {
  "venue_name": "Bull McCabes",
  "city": "Durham",
  "state": "NC",
  "address": "123 Main st.",
  "zipcode": "27705",
  "venue_type": "Bar",
  "id": 1,
  "avgPrice": "2.2500000000000000",
  "avgRating": "3.2500000000000000",
  "avgVolume": "3.5000000000000000"
}

it('renders Result without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <Result venue={venue} />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders Result as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <Result name="Result" venue={venue}/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
