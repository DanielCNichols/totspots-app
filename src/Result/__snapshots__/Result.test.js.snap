// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Result as expected 1`] = `
<li
  className="result"
  onClick={[Function]}
>
  <div
    className="result_rating"
  >
    <h3>
      Bull McCabes
    </h3>
    <span
      className="rating"
    >
      <div>
        <span>
          ★
        </span>
        <span>
          ★
        </span>
        <span>
          ★
        </span>
      </div>
    </span>
  </div>
  <div
    className="result_address"
  >
    <span>
      Bar
    </span>
    <div
      className="price_span"
    >
      <span>
        $
      </span>
      <span>
        $
      </span>
    </div>
    <p>
      Phone: 
      Not provided
    </p>
    <p>
      <span>
        Address: 
        123 Main st.
      </span>
       
      <span
        className="city"
      >
        Durham
      </span>
      ,
      <span
        className="state"
      >
         
        NC
      </span>
    </p>
    <p>
      Website: 
      Not provided
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    See more
  </button>
</li>
`;
